toolkit/tools/pkg/imagecustomizerlib/customizepartitionsfilecopy.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func customizePartitionsUsingFileCopy(buildDir string, baseConfigPath string, config *imagecustomizerapi.Config,
+	buildImageFile string, newBuildImageFile string,
+) error {
+	existingImageConnection, err := connectToExistingImage(buildImageFile, buildDir, "imageroot")
+	if err != nil {
+		return err
+	}
+	defer existingImageConnection.Close()
+
+	diskConfig := (*config.Disks)[0]
+
+	installOSFunc := func(imageChroot *safechroot.Chroot) error {
+		return copyFilesIntoNewDisk(existingImageConnection.Chroot(), imageChroot)
+	}
+
+	newImageConnection, err := createNewImage(newBuildImageFile, diskConfig, config.SystemConfig.PartitionSettings,
+		config.SystemConfig.BootType, buildDir, "newimageroot", installOSFunc)
+	if err != nil {
+		return err
+	}
+	defer newImageConnection.Close()
+
+	err = newImageConnection.CleanClose()
+	if err != nil {
+		return err
+	}
+
+	err = existingImageConnection.CleanClose()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyFilesIntoNewDisk(existingImageChroot *safechroot.Chroot, newImageChroot *safechroot.Chroot) error {
+	err := copyFilesIntoNewDiskHelper(existingImageChroot, newImageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to copy files into new partition layout:\n%w", err)
+	}
+	return nil
+}
+
+func copyFilesIntoNewDiskHelper(existingImageChroot *safechroot.Chroot, newImageChroot *safechroot.Chroot) error {
+	copyArgs := []string{"--verbose", "--no-clobber", "-a", "--no-dereference", "--sparse", "always", "-t", newImageChroot.RootDir()}
+
+	files, err := os.ReadDir(existingImageChroot.RootDir())
+	if err != nil {
+		return fmt.Errorf("failed to read base image root directory:\n%w", err)
+	}
+
+	for _, file := range files {
+		switch file.Name() {
+		case "dev", "proc", "sys", "run", "tmp":
```

Comment:
afaik some dirs under /var/ might not be meaningful to be copied over to the new disk? do we need to list them all here?

Reply:
Those are a couple of symlinks under `/var`, including `/var/lock` and `/var/run`. Since `cp` is called with `--no-dereference`, these symlinks will be copied as symlinks.

Are there any other directories you are concerned about?

Reply:
got it. no other concerns